src/peergos/server/net/MultipartReceiver.java

```diff
@@ -0,0 +1,46 @@
+package peergos.server.net;
+
+import java.io.*;
+import java.util.*;
+
+public class MultipartReceiver {
+
+    public static byte[] extractFile(InputStream in, String boundary) {
+        try {
+            String first = readLine(in);
+            if (!first.substring(2).equals(boundary))
+                throw new IllegalStateException("Incorrect boundary! " + boundary + " != " + first.substring(2));
+            String filenameLine = readLine(in);
+            String contentType = readLine(in);
```

Comment:
This looks  super useful  and re-usable. Could we add a test for the basic functionality and error conditions?